### PR TITLE
Porting more APIs to 0.3

### PR DESCRIPTION
**Pull request by Susurrus**
Re-submission of #438 since it was accidentally closed.

### CODE REVIEW

**Comment by Susurrus** [comment]
@retep998 Alright, this has been rebased and should be good-to-go.

File: src/shared/sspi.rs [hunk 1]
```diff
@@ -4,5 +4,1220 @@
...
+pub type PSEC_WINNT_AUTH_DATA = *mut SEC_WINNT_AUTH_DATA;
+
+STRUCT!{struct SEC_WINNT_AUTH_PACKED_CREDENTIALS {
+    cbHeaderLength: c_ushort,
+    cbStructureLength: c_ushort,
+    AuthData: SEC_WINNT_AUTH_DATA,
+}}
+pub type PSEC_WINNT_AUTH_PACKED_CREDENTIALS = *mut SEC_WINNT_AUTH_PACKED_CREDENTIALS;
+
+DEFINE_GUID!(SEC_WINNT_AUTH_DATA_TYPE_PASSWORD,
```
Comment (retep998) [inline src/shared/sspi.rs]: Wrong formatting for GUIDs.

File: src/um/processthreadsapi.rs [hunk 2]
```diff
@@ -60,31 +68,32 @@ STRUCT!{struct STARTUPINFOW {
     hStdError: HANDLE,
 }}
 pub type LPSTARTUPINFOW = *mut STARTUPINFOW;
+
 extern "system" {
     pub fn QueueUserAPC(
         pfnAPC: PAPCFUNC,
         hThread: HANDLE,
-        dwData: ULONG_PTR,
+        dwData: ULONG_PTR
```
Comment (retep998) [inline src/um/processthreadsapi.rs]: Why are you getting rid of trailing commas?

Comment (Susurrus) [inline src/um/processthreadsapi.rs]: This is done automatically by rustfmt when I was experimenting with it. I think I brought this up in gitter, but I can't remember. I don't see any specific comment on it in CONTRIBUTING. You want trailing commas?

Comment (retep998) [inline src/um/processthreadsapi.rs]: https://github.com/retep998/winapi-rs/blob/dev/CONTRIBUTING.md#extern-functions has a trailing comma. Macros for functions (like `RIDL!` and `FN!`) *require* trailing commas. So yes, trailing commas for all functions.

**Comment by Susurrus** [comment]
I've gone through this whole commit and should have fixed all of the styling for `DEFINE_GUID!` and trailing commas for functions.

**Comment by Susurrus** [comment]
Went through again to remove all the blank lines you don't like (except around the import statements so it's easier to see where the code actually starts) and found a few more missing trailing commas.

File: src/um/subauth.rs [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+pub const USER_NO_AUTH_DATA_REQUIRED: ULONG = 0x00080000;
+pub const USER_PARTIAL_SECRETS_ACCOUNT: ULONG = 0x00100000;
+pub const USER_USE_AES_KEYS: ULONG = 0x00200000;
+pub const NEXT_FREE_ACCOUNT_CONTROL_BIT: ULONG = USER_USE_AES_KEYS << 1;
+pub const USER_MACHINE_ACCOUNT_MASK: ULONG = USER_INTERDOMAIN_TRUST_ACCOUNT
+    | USER_WORKSTATION_TRUST_ACCOUNT | USER_SERVER_TRUST_ACCOUNT;
+pub const USER_ACCOUNT_TYPE_MASK: ULONG = USER_TEMP_DUPLICATE_ACCOUNT | USER_NORMAL_ACCOUNT
+    | USER_MACHINE_ACCOUNT_MASK;
+pub const USER_COMPUTED_ACCOUNT_CONTROL_BITS: ULONG = USER_ACCOUNT_AUTO_LOCKED
+| USER_PASSWORD_EXPIRED;
```
Comment (retep998) [inline src/um/subauth.rs]: Missing indent.

Comment (Susurrus) [inline src/um/subauth.rs]: Fixed.

**Comment by Susurrus** [comment]
Thanks, @retep998!

**Pull request merged**